### PR TITLE
Update JITaaS option names

### PR DESCRIPTION
**Pull request by a7ehuo**
[skip ci]
Based on the decision from #6203, replace JITaaS option names as below:
`-XX:JITaaSServer` => `-XX:+StartAsJITServer`
`-XX:JITaaSClient` => `-XX:+UseJITServer`

Issue #6203

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by mpirvu** [comment]
The options need to have a `+` in front of them. E.g.: `-XX:+StartAsJITServer` `-XX:+UseJITServer`

File: doc/compiler/jitaas/Usage.md [hunk 1]
```diff
@@ -20,38 +20,38 @@ OpenJDK Assembly Exception [2].
 SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
 -->
 
-JITaaS adds an additional two *personas* to the `java` executable: client mode and server mode. There are two new command line options that can be used to select a persona: `-XX:JITaaSServer` and `-XX:JITaaSClient`.
+JITaaS adds an additional two *personas* to the `java` executable: client mode and server mode. There are two new command line options that can be used to select a persona: `-XX:+StartAsJITServer` and `-XX:+UseJITServer`.
```
Comment (mpirvu) [inline doc/compiler/jitaas/Usage.md]: I think we should delete the mention of JITaaS

File: runtime/compiler/control/J9Options.cpp [hunk 2]
```diff
@@ -2028,30 +2028,30 @@ J9::Options::fePreProcess(void * base)
    if (!TR::Compiler->target.cpu.isZ())
       self()->setOption(TR_DisableAOTBytesCompression);
 
-   // Check for option -XX:JITaaSClient and/or -XX:JITaaSServer
+   // Check for option -XX:+UseJITServer and/or -XX:+StartAsJITServer
```
Comment (mpirvu) [inline runtime/compiler/control/J9Options.cpp]: In theory one could have the options with "-" as well (meaning I want to disable a feature that was set previously). Could we take into account those as well?
You would first determine if client mode is enabled by comparing the index of "-XX:-UseJITServer" to the index "-XX:+UseJITServer". Then do the same for server mode. If both the client and server mode are set, then compare the index of "-XX:+UseJITServer" to the index of "-XX:+StartAsJITServer"

File: runtime/compiler/control/CompilationRuntime.hpp [hunk 3]
```diff
@@ -439,6 +439,9 @@ class CompilationInfo
          }
       }; // CompilationStatsPerInterval
 
+   static bool isJITServerEnabled() { return _isJITServerEnabled; }
```
Comment (mpirvu) [inline runtime/compiler/control/CompilationRuntime.hpp]: How are these methods going to be used? Why isn't enough the mode we set in persistentInfo?

File: doc/compiler/jitaas/Usage.md (hunk 1 above)
Comment (a7ehuo) [reply doc/compiler/jitaas/Usage.md]: Will replace 
`JITaaS` => `JITServer` in all documents under `doc/compiler/jitaas` at where it applies. 
`doc/compiler/jitaas`=> `doc/compiler/jitserver`

File: runtime/compiler/control/CompilationRuntime.hpp (hunk 3 above)
Comment (a7ehuo) [reply runtime/compiler/control/CompilationRuntime.hpp]: I thought we needed another runtime check. Will remove it.

File: doc/compiler/jitaas/Usage.md (hunk 1 above)
Comment (a7ehuo) [reply doc/compiler/jitaas/Usage.md]: will leave the directory name change out from this PR for now for easier diff comparison.

**Comment by a7ehuo** [comment]
@mpirvu Updated and ready for review. Thanks Marius!

File: runtime/compiler/control/J9Options.cpp [hunk 4]
```diff
@@ -2028,30 +2039,40 @@ J9::Options::fePreProcess(void * base)
...
+      char *xxDisableStartAsJITServerOption = "-XX:-StartAsJITServer";
+
+      int32_t xxUseJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxUseJITServerOption, 0);
+      int32_t xxDisableUseJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxDisableUseJITServerOption, 0);
+      int32_t xxStartAsJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxStartAsJITServerOption, 0);
+      int32_t xxDisableStartAsJITServerArgIndex = FIND_ARG_IN_VMARGS(STARTSWITH_MATCH, xxDisableStartAsJITServerOption, 0);
+
       // Check if option is at all specified
-      if (xxJITaaSClientArgIndex >= 0 || xxJITaaSServerArgIndex >= 0)
+      if (((xxUseJITServerArgIndex > xxDisableUseJITServerArgIndex ) && (xxUseJITServerArgIndex >= 0)) ||
```
Comment (mpirvu) [inline runtime/compiler/control/J9Options.cpp]: There is an extra space before closing parenthesis.
Do you need the check for  `(xxUseJITServerArgIndex >= 0)`? I am guessing that if no option is present then `xxUseJITServerArgIndex` has the same value as `xxDisableUseJITServerArgIndex` and the part after `&&` does not get evaluated.

Comment (a7ehuo) [reply runtime/compiler/control/J9Options.cpp]: You're right. No need to compare `(xxUseJITServerArgIndex >= 0)`

**Comment by a7ehuo** [comment]
@mpirvu Updated. Ready for Review. Thanks!

**Comment by mpirvu** [comment]
Jenkins test sanity.functional+jitaas xlinux jdk8 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
```
22:52:02  FAILED test targets:
22:52:02  	cmdLineTester_gcsuballoctests_0
```
Can we verify that this fails even without JITaaS?

**Comment by mpirvu** [comment]
Jenkins test sanity.functional jdk8 depends eclipse/openj9-omr#jitaas

**Comment by mpirvu** [comment]
Jenkins test sanity.functional xlinux jdk8 depends eclipse/openj9-omr#jitaas

**Comment by a7ehuo** [comment]
Looks like without `jitaas`, `cmdLineTester_gcsuballoctests_0` still fails
```
12:10:28  FAILED test targets:
12:10:28  	cmdLineTester_gcsuballoctests_0
```

**Comment by mpirvu** [comment]
Test error is not due to this change so we can merge the code.

**Pull request merged**